libcodechecker/server.py

```diff
@@ -0,0 +1,490 @@
+    """
+
+    return {
+        'prog': 'CodeChecker server',
+        'formatter_class': argparse.ArgumentDefaultsHelpFormatter,
+
+        # Description is shown when the command's help is queried directly
+        'description': "The CodeChecker Web server is used to navigate "
+                       "analysis results. A started server can be connected "
+                       "to via a Web browser, or by using the 'CodeChecker "
+                       "cmd' command-line client.",
+
+        # Help is shown when the "parent" CodeChecker command lists the
+        # individual subcommands.
+        'help': "Start and manage the CodeChecker Web server"
+    }
+
+
+def add_arguments_to_parser(parser):
+    """
+    Add the subcommand's arguments to the given argparse.ArgumentParser.
+    """
+
+    # TODO: --workspace is an outdated concept in 'store'. Later on,
+    # it shall be deprecated, as changes to db_handler commence.
+    parser.add_argument('-w', '--workspace',
+                        type=str,
+                        dest="workspace",
+                        default=util.get_default_workspace(),
+                        required=False,
+                        help="Directory where CodeChecker can store analysis "
+                             "result related data, such as the database. "
+                             "(Cannot be specified at the same time with "
+                             "'--sqlite' or '--config-directory'.)")
+
+    parser.add_argument('-f', '--config-directory',
+                        type=str,
+                        dest="config_directory",
+                        default=util.get_default_workspace(),
+                        required=False,
+                        help="Directory where CodeChecker server should read "
+                             "server-specific configuration (such as "
+                             "authentication settings) from.")
+
+    parser.add_argument('--host',
+                        type=str,
+                        dest="listen_address",
+                        default="localhost",
+                        required=False,
+                        help="The IP address or hostname of the server on "
+                             "which it should listen for connections.")
+
+    # TODO: -v/--view-port is too verbose. The server's -p/--port is used
+    # symmetrically in 'CodeChecker cmd' anyways.
+    parser.add_argument('-v', '--view-port',  # TODO: <- Deprecate and remove.
+                        '-p', '--port',
+                        type=int,
+                        dest="view_port",
+                        metavar='PORT',
+                        default=8001,
+                        required=False,
+                        help="The port which will be used as listen port for "
+                             "the server.")
+
+    # TODO: This should be removed later on, in favour of --listen-address.
```

Comment:
I'm not sure we want to keep the `--listen-address` I think `--host 0.0.0.0` will mean the same configuration as the `--not-host-only` right now.